### PR TITLE
Display banner to enable 2FA when setting up VBBA

### PR DESCRIPTION
**Pull request by MonilBhavsar**
### Details

- Display 2FA prompt when adding bank account during verification and validation steps.
- Redirects user to olddot, once 2FA setup is completed, redirects them back to newdot.

### Fixed Issues

https://github.com/Expensify/Expensify/issues/238668
PROPOSAL: N/A

### Tests

Let user have 2FA disabled
Tests with https://github.com/Expensify/Web-Expensify/pull/35394.
1. Checkout that branch
2. Run app
3. Change the URL to `http://localhost:8080/bank-account`
4. Go till Validation step and confirm the 2FA prompt
5. Click on secure account and confirm it redirects to olddot, enables 2FA. 
6. Finish the setup and confirm it redirects back to newdot and this time it has no prompt
7. Go to olddot and disable 2FA
8. I manually changed account state, to go through validation state and confirmed the 2FA prompt there

- [x] Verify that no errors appear in the JS console

### QA Steps

Prerequisite: User should have 2FA **disabled** in olddot > Settings > Accounts

1. Start the process of setting up bank account in VERIFYING state
2. Go till Verifying state and confirm the 2FA prompt as in screenshot
3. Click on "Secure Account" and confirm you're redirected to olddot and 2FA modal is open.
4. Finish the 2FA setup. Upon clicking Finish, confirm you're redirected back to newdot
5. Confirm this time there is not any 2FA prompt.

Now, 

6. Disable 2FA from olddot
7. Go to bank account setup again. Can go to URL `staging.new.expensify.com/bank-account/`
8. Wait till Bank account is verified and you go to validation step and see the inputs to enter 3 amounts
9. Confirm 2FA prompt as in screenshot
10. Click on "Secure Account" and confirm you're redirected to olddot and 2FA modal is open.
11. Finish the 2FA setup. Upon clicking Finish, confirm you're redirected back to newdot
12. Confirm this time there is not any 2FA prompt.

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

Screenshot 2022-12-01 at 5 43 31 PM
Screenshot 2022-12-01 at 5 46 15 PM

</details>

<details>
<summary>Mobile Web - Chrome</summary>

</details>

<details>
<summary>Mobile Web - Safari</summary>

</details>

<details>
<summary>Desktop</summary>

Screenshot 2022-12-01 at 5 50 33 PM
Screenshot 2022-12-01 at 5 51 00 PM

</details>

<details>
<summary>iOS</summary>

</details>

<details>
<summary>Android</summary>

</details>

### CODE REVIEW

**Comment by MonilBhavsar** [comment]
Please hold reviewing for now. Was supposed to be draft. Need to update some things

**Comment by MonilBhavsar** [comment]
This is ready for review now 👍

**Review by shawnborton — DISMISSED** [review]

**Comment by MonilBhavsar** [comment]
@rushatgabhane fixed

**Comment by MonilBhavsar** [comment]
@rushatgabhane are you getting to validation and verification step now?
You'll be able to test redirection on staging after deployment (few hours)

**Comment by MonilBhavsar** [comment]
Ah I see, doesn't the test bank account skip onfido flow?

**Comment by MonilBhavsar** [comment]
Also this is off HOLD

**Comment by MonilBhavsar** [comment]
I asked but as per the [discussion](https://expensify.slack.com/archives/C01SKUP7QR0/p1668678556126539), @pecanoro you'll need to test it

**Comment by pecanoro** [comment]
Sorry for the delay, I will test this tomorrow, can you in the meantime solve the conflicts, please?

**Comment by MonilBhavsar** [comment]
@shawnborton I see icons have changed at most of the places. Should we use new icon or use the old one only.

**Comment by pecanoro** [comment]
@MonilBhavsar Can you run the last commit? It's giving a lot of trouble when running the app

**Comment by MonilBhavsar** [comment]
Updated main, but holding on if we want to use a new icon for dark theme

**Comment by pecanoro** [comment]
@rushatgabhane Can you complete the PR reviewer list for the things you can test and review? I will add the screenshots and such from testing.

**Comment by shawnborton** [comment]
@MonilBhavsar yes, let's use a new icon here. Here ya go: [simple-illustration__shield.svg.zip](https://github.com/Expensify/App/files/10116190/simple-illustration__shield.svg.zip)


You can add it to this folder: https://github.com/Expensify/App/tree/main/assets/images/simple-illustrations

File: src/languages/es.js
Comment (pecanoro) [inline src/languages/es.js:797]:
```suggestion
        enable2FAText: 'Tu seguridad es importante para nosotros, por favor configura ahora la autenticación de dos factores. Eso nos permitirá disputar las transacciones de la Tarjeta Expensify y reducirá tu riesgo de fraude.',
```

File: src/languages/es.js
Comment (pecanoro) [inline src/languages/es.js:798]:
```suggestion
        secureYourAccount: 'Asegura tu cuenta',
```

**Review by pecanoro — CHANGES_REQUESTED** [review]

**Comment by pecanoro** [comment]
Something seems broken with adding a new bank account, I can't pass the company steps because I get this [error](https://github.com/Expensify/Web-Expensify/blob/43b577f6c67618b2465d6f362615a243a8e559f2/lib/BankAccountAPI.php#L1178). The routing and account number are not being passed to the API and it's blocking the verification flow. @MonilBhavsar Can you check if it's just me?

**Comment by pecanoro** [comment]
Ok, I will try retest everything again now that we solved the deploy blocker.

File: src/pages/ReimbursementAccount/ValidationStep.js
Comment (pecanoro) [inline src/pages/ReimbursementAccount/ValidationStep.js:39]: The guidelines mention: "All props are defined accurately and each prop has a /** comment above it */"

**Review by pecanoro — CHANGES_REQUESTED** [review]

**Comment by pecanoro** [comment]
You also need to merge with main so you can retest the PR after all the changes.

**Comment by pecanoro** [comment]
Ok, I completed the checklist, it works pretty well. Please address the comment, update the screenshots after merging with main and I will merge it tomorrow!!

**Comment by pecanoro** [comment]
@MonilBhavsar Something that is not fully working for me but it might just my emulator is opening the link from mobile safari to set up 2FA on oldDot, it's not redirecting. Can you check?

**Comment by MonilBhavsar** [comment]
You'll need to go to Settings > Safari > and disable "block pop-ups"

**Comment by MonilBhavsar** [comment]
Addressed reviews and merged main

**Comment by MonilBhavsar** [comment]
Updated screenshots with dark theme cc @shawnborton

**Review by shawnborton — APPROVED** [review]

**Comment by shawnborton** [comment]
Looks good!

**Review by pecanoro — APPROVED** [review]

**Comment by pecanoro** [comment]
@rushatgabhane Ready for your final review and I will merge after!!

**Review by rushatgabhane — APPROVED** [review]
@pecanoro good to merge!

**Comment by francoisl** [comment]
Hey @MonilBhavsar, @mvtglobally / @kavimuru  and I were unable to QA this on staging, and with different errors. After enabling 2FA, they get redirected back to the "Connect Bank Account Page" (see https://github.com/Expensify/App/issues/13326#issuecomment-1341338294), whereas the NewDot shows a loading spinner forever for me, on mWeb iOS & Android:

https://user-images.githubusercontent.com/2229301/206257867-b136eb38-3e87-4d37-9543-7f9c951da102.mov

Any idea what could be going on?

**Comment by MonilBhavsar** [comment]
I'm also seeing this issue on web. Looks like something has changed. I think we'll need to specify the route more specifically.

File: src/pages/ReimbursementAccount/Enable2FAPrompt.js
Comment (eVoloshchak) [inline src/pages/ReimbursementAccount/Enable2FAPrompt.js:30]: 👋 Heads-up that this has caused a small visual issue in https://github.com/Expensify/App/issues/20932
This icon wasn't changing color on hover, since we were overriding its color with `iconFill` prop

**Pull request merged**
